tools/codegen/gen.py

```diff
@@ -656,7 +661,7 @@ def compute_registration_declarations(f: NativeFunction) -> str:
         'schema': f'aten::{f.func}',
         # TODO: What exactly is the semantics of the 'dispatch' field?
         'dispatch': str(f.dispatch.keys() != {DispatchKey.CompositeImplicitAutograd}),
-        'default': str(any(is_generic_dispatch_key(k) for k in f.dispatch))
+        'default': str(any(is_generic_dispatch_key(k) for k in f.dispatch) or has_autogenerated_composite_kernel(f))
```

Comment:
I thought about maybe just putting the `CompositeImplicitAutograd` key directly in each NativeFunction's dispatch table in `model.py`, so we wouldn't need this logic here. But that has some downsides: e.g. there's some code in `register_dispatch_key.py` that would like to be able to differentiate "real" CompositeImplicitAutograd kernels from ones that we've generated, which would become harder to do.

Reply:
With your upcoming rewrite this probably should be moot soon, so I don't mind this special case.